Marv-Lous/Marv-Lous/Controllers/LoginPageViewController.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  LoginPageViewController.swift
+//  Marv-Lous
+//
+//  Created by Erik Egers on 2022/04/01.
+//
+
+import Foundation
+import UIKit
+
+class LoginPageViewController: UIViewController {
+    
+    private lazy var loginPageViewModel = LoginPageViewModel(delegate: self)
+    
+    @IBOutlet private weak var usernameTextField: UITextField!
+    @IBOutlet private weak var passwordTextField: UITextField!
+    @IBOutlet weak var loginErrorDescriptionLabel: UILabel!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTextFields()
+    }
+    
+    @IBAction private func loginButtonPressed(_ sender: UIButton) {
+        let username = usernameTextField.text ?? ""
+        let password = passwordTextField.text ?? ""
+        
+        loginPageViewModel.attemptLogin(username, password)
+    }
+    
+    func setupTextFields() {
+        usernameTextField.delegate = self
+        passwordTextField.delegate = self
+    }
+}
+
+// MARK: - Text Field Delegate Methods
+extension LoginPageViewController: UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        usernameTextField.resignFirstResponder()
```

Comment:
This is just a suggestion but is not better to rather show an alert notifying the user they have put in their username and password incorrectly instead of showing an error description label. This approach you have taken is not native to IOS it is more android or web driven.

Reply:
I implemented the alert (see above gif).